multisig-wallet/project/contracts/multisig-contract/src/main.sw

```diff
@@ -82,69 +81,61 @@ impl MultiSignatureWallet for Contract {
-                },
-            };
-
-            if transfer_params.value.is_some() {
-                require(transfer_params.value.unwrap() <= this_balance(transfer_params.asset_id), ExecutionError::InsufficientAssetAmount);
+        match transaction_parameters {
+            TransactionParameters::Call(contract_call_params) => {
+                let target_contract_id = match target {
+                    Identity::ContractId(contract_identifier) => contract_identifier,
+                    _ => {
+                        log(ExecutionError::CanOnlyCallContracts);
+                        revert(FAILED_REQUIRE_SIGNAL)
+                    },
+                };
+
+                if contract_call_params.transfer_params.value.is_some() {
+                    require(contract_call_params.transfer_params.value.unwrap() <= this_balance(contract_call_params.transfer_params.asset_id), ExecutionError::InsufficientAssetAmount);
+                }
+
+                let transaction_hash = compute_hash(TypeToHash::Transaction(Transaction::new(contract_id(), nonce, target, transaction_parameters)));
+                let approval_count = count_approvals(signatures, transaction_hash);
+                require(storage.threshold.read() <= approval_count, ExecutionError::InsufficientApprovals);
+
+                storage.nonce.write(nonce + 1);
+
+                let call_params = CallParams {
+                    coins: contract_call_params.transfer_params.value.unwrap_or(0),
+                    asset_id: contract_call_params.transfer_params.asset_id,
+                    gas: contract_call_params.forwarded_gas,
+                };
+                call_with_function_selector(target_contract_id, contract_call_params.function_selector, contract_call_params.calldata, contract_call_params.single_value_type_arg, call_params);
+
+                log(ExecuteTransactionEvent { nonce, target
+                // transaction_parameters,// TODO: Uncomment when SDK supports logs with nested Bytes https://github.com/FuelLabs/fuels-rs/issues/1046
+ });
+            },
+            TransactionParameters::Transfer(transfer_params) => {
+                require(transfer_params.value.is_some(), ExecutionError::TransferRequiresAValue);
+                let value = transfer_params.value.unwrap();
+                require(value <= this_balance(transfer_params.asset_id), ExecutionError::InsufficientAssetAmount);
+
+                let transaction_hash = compute_hash(TypeToHash::Transaction(Transaction::new(contract_id(), nonce, target, transaction_parameters)));
+                let approval_count = count_approvals(signatures, transaction_hash);
+                require(storage.threshold.read() <= approval_count, ExecutionError::InsufficientApprovals);
+
+                storage.nonce.write(nonce + 1);
+
+                transfer(value, transfer_params.asset_id, target);
+
+                log(ExecuteTransactionEvent { nonce, target
```

Comment:
Might as well move the log to be outside of the match expression, at the end of the function